BlazarBase/src/main/java/com/hubspot/blazar/base/branch/BranchStatus.java

```diff
@@ -0,0 +1,86 @@
+package com.hubspot.blazar.base.branch;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+import com.hubspot.blazar.base.GitInfo;
+import com.hubspot.blazar.base.MalformedFile;
+import com.hubspot.blazar.base.ModuleState;
+import com.hubspot.blazar.base.RepositoryBuild;
+
+public class BranchStatus {
+  private final List<RepositoryBuild> queuedBuilds;
+  private final Set<ModuleState> moduleStates;
+  private final Set<GitInfo> otherBranches;
+  private Set<MalformedFile> malformedFiles;
+  private GitInfo branch;
+
+  @JsonCreator
+  public BranchStatus(@JsonProperty("queuedBuilds") List<RepositoryBuild> queuedBuilds,
```

Comment:
@gchomatas this is the new object that contains _all_ the data that will go on the new module-centric page.

- `queuedBuilds` are repoBuilds that have yet to be started.
- `moduleStates` is the data we are using to populate the cards on the page.
- `otherBranches` is the data for the dropdown that shows what other branches can be viewed on this repository.
- `malformedFiles` the malformed configuration files that have been detected on this branch.
- `gitInfo` The branch information (name etc. so that @andyhuang91 does not have to make an additional query for this data)